### PR TITLE
Add overrides for default base directory

### PR DESCRIPTION
**Pull request by natemcmaster**
cc @bricelam

### CODE REVIEW

File: src/Microsoft.Data.Sqlite/SqliteConnection.cs [hunk 1]
```diff
@@ -211,7 +211,8 @@ private static string BaseDirectory
             {
                 var appDataType = CurrentApplicationData?.GetType();
                 var localFolder = appDataType?.GetRuntimeProperty("LocalFolder").GetValue(CurrentApplicationData);
-                return (localFolder?.GetType().GetRuntimeProperty("Path").GetValue(localFolder) as string)
+                return Environment.GetEnvironmentVariable("ADONET_DATA_DIR")
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: We need this on `net451` too (a few lines down)

Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: Updated.

**Comment by bricelam** [comment]
:shipit:

**Comment by davidfowl** [comment]
Can this be settable via a first class API as well?

**Comment by bricelam** [comment]
@davidfowl Wouldn't you just use a full path if you wanted that?

**Comment by natemcmaster** [comment]
This is really just a workaround for https://github.com/aspnet/EntityFramework/issues/4765, not really for general use.

**Pull request merged**